apps/freeablo/faworld/faction.cpp

```diff
@@ -0,0 +1,21 @@
+#include "faction.h"
+
+#include "actor.h"
+
+namespace FAWorld
+{
+
+    bool Faction::canAttack(Faction* other)
+    {
+        // Attacker -> Attackee
+        static bool ruleTable[FactionType::ENUM_END][FactionType::ENUM_END] =
```

Comment:
Sounds good, but then we need to have townsfolk in some faction and we don't want the player attack them.  So we can file them under hero, or just have two alliances now "heaven" and "hell" for people and monsters respectively.